### PR TITLE
Force the initialization of throwMethod in Definitions

### PR DESCRIPTION
**Pull request by VladimirNik**
This change will fix the problem with unpickling of throw expression from TASTY representation.

### CODE REVIEW

**Comment by DarkDimius** [comment]
LGTM after tests pass

**Comment by VladimirNik** [comment]
Review by @odersky , @DarkDimius

File: src/dotty/tools/dotc/core/Definitions.scala [hunk 1]
```diff
@@ -785,8 +785,8 @@ class Definitions {
     EmptyPackageVal,
     OpsPackageClass)
 
-    /** Lists core methods that don't have underlying bytecode, but are synthesized on-the-fly in every reflection universe */
-    lazy val syntheticCoreMethods = AnyMethods ++ ObjectMethods ++ List(String_+)
+  /** Lists core methods that don't have underlying bytecode, but are synthesized on-the-fly in every reflection universe */
+  lazy val syntheticCoreMethods = AnyMethods ++ ObjectMethods ++ List(String_+, throwMethod)
```
Comment (DarkDimius) [inline src/dotty/tools/dotc/core/Definitions.scala]: indentation here was intentional: member definitions are indented with 2 more spaces compared to class definitions.

**Pull request merged**